cmd/state-svc/internal/rtwatcher/watcher.go

```diff
@@ -100,6 +102,32 @@ func (w *Watcher) RecordUsage(e entry) {
 	w.an.EventWithSource(anaConst.CatRuntimeUsage, anaConst.ActRuntimeHeartbeat, e.Source, e.Dims)
 }
 
+// GetProcessesInUse returns for a given runtime exec directory a map of running executables to
+// their process IDs.
+func (w *Watcher) GetProcessesInUse(execDir string) map[string]int {
+	inUse := map[string]int{}
+
+	if runtime.GOOS != "linux" {
+		execDir = strings.ToLower(execDir) // Windows and macOS filesystems are case-insensitive
+	}
+	for _, proc := range w.watching {
+		exeToCompare := proc.Exec
+		if runtime.GOOS != "linux" {
+			exeToCompare = strings.ToLower(exeToCompare) // Windows and macOS filesystems are case-insensitive
+		}
+		if !strings.Contains(exeToCompare, execDir) {
+			continue
+		}
+		if isRunning, err := proc.IsRunning(); err == nil && isRunning {
+			inUse[proc.Exec] = proc.PID
+		} else if err != nil {
+			multilog.Error("Could not check if runtime process is running: %s", errs.JoinMessage(err))
```

Comment:
I don't think this is a good idea because the watcher may be watching multiple processes. As long as one of them is in use, we need to know. Otherwise, returning an error prematurely will result in an "all clear" as far as the returned result is concerned.

Reply:
Since I removed the explicit `IsRunning()` check, there's no need for an error check.

Reply:
I'm still struggling over this a little. I feel like our rollbar is gonna get spammed with this due to errors that effectively amount to user input of some sort. Suggest we turn this into a `logging.Warning()`.

Reply:
We haven't been getting spammed by https://github.com/ActiveState/cli/blob/5a71fb6fc2c1b4afff59987813ee11a9a4ebc5da/cmd/state-svc/internal/rtwatcher/watcher.go#L84, so I think it's pretty safe. I will add `logging.Warning()` if you insist, though.